draft-dt-tls-external-psk-guidance.md

```diff
@@ -312,11 +289,20 @@ are validate to be between \[1, 16\] bytes.
+Section 5.1 of {{?RFC4279}} mandates that the PSK identity should be first converted to a character string and then encoded to octets using UTF-8. This was done to avoid interoperability problems (especially when the identity is configured by human users). On the other hand, {{?RFC7925}} advises  implementations against assuming any structured format for PSK identities and recommends byte-by-byte comparison for any operation. TLS version 1.3 {{!RFC8446}} follows the same practice of specifying the PSK identity as a sequence of opaque bytes (shown as opaque identity<1..2^16-1> in the specification). {{?RFC8446}} also requires that the PSK identities are at least 1 byte and at the most 65535 bytes in length. Although {{!RFC8446}} does not place strict requirements on the format of PSK identities, we do however note that the format of PSK identities can vary depending on the deployment:
+
+- The PSK identity MAY be a user configured string when used in protocols like Extensible Authentication Protocol (EAP) {{?RFC3748}}. gnuTLS for example treats PSK identities as usernames.
+- PSK identities MAY have a domain name suffix for roaming and federation. 
+- Deployments should take care that the length of the PSK identity is sufficient to avoid obvious collisions. {{identity-collision}} discusses some implication of collisions between external and resumption PSKs.
+
+
+#### PSK Identity collisions {#identity-collision}
+
+# Security Considerations {#security-con}
 
-Other assumptions unique to different stacks are listed below.
+It is NOT RECOMMENDED to share the same PSK between more than one client and server. However, as discussed in {{use-cases}}, there are application scenarios that may rely on sharing the same PSK among multiple nodes. {{!I-D.ietf-tls-external-psk-importer}} helps in mitigating selfie style reflection attacks if a pair of client and server can switch roles. This is achieved by correctly using the node identifiers in the ImportedIdentity.context construct as explained in Appendix B of {{!I-D.ietf-tls-external-psk-importer}}. When a PSK is shared between a client and server that can switch roles, it is not necessary to know the identifier(s) of the other endpoint. Instead, it is sufficient to check that the identifier of the other endpoint in context is not equal to any of one’s own identifiers. To simplify implementation of this check, it is RECOMMENDED that each endpoint selects one globally unique identifier and uses it in all PSK handshakes. The unique identifier can, for example, be one of its MAC addresses or a 32-byte random number. Alternatively, nodes can also use a Universally Unique IDentifier (UUID) URN {{?RFC4122}} as their identifier. Including node identifiers in the ImportedIdentity.context construct also ensures that condition (2) is met.
```

Comment:
As in my other comment, I think this is too much text to get the point across. It needs to be simplified

Reply:
I have shortened the security considerations section somewhat. We can further shorten it if others also feel the same way.

Reply:
@chris-wood made another commit that further simplifies the text.